### PR TITLE
Add ssl support to URI parser

### PR DESCRIPTION
**Pull request by chefabien**
CreateFromURI method from Channel object does not currently handle SSL connections. This adds a new CreateFromURI that takes SSL parameters.

### CODE REVIEW

File: src/Channel.cpp [hunk 1]
```diff
@@ -90,6 +90,47 @@ Channel::ptr_t Channel::CreateFromUri(const std::string &uri, int frame_max)
...
+    amqp_default_connection_info(&info);
+
+    boost::shared_ptr<char> uri_dup = boost::shared_ptr<char>(strdup(uri.c_str()), free);
+
+    if (0 != amqp_parse_url(uri_dup.get(), &info))
+    {
+        throw BadUriException();
+    }
+
+    if ( info.ssl )
```
Comment (alanxz) [inline src/Channel.cpp]: This function should only accept amqps:// URIs, it should throw something if a non-secure URI is detected.

File: src/Channel.cpp [hunk 2]
```diff
@@ -90,6 +90,47 @@ Channel::ptr_t Channel::CreateFromUri(const std::string &uri, int frame_max)
                   frame_max);
 }
 
+Channel::ptr_t Channel::CreateFromUri(const std::string &uri,
```
Comment (alanxz) [inline src/Channel.cpp]: Lets call this CreateSecureFromUri

**Comment by alanxz** [comment]
As noted in the comments above, I'd like to make this API have less chance that bad input won't do something unexpected.  The case I'm worrying about is someone mis-typing in a configuration dropping the s in amqps://... and not having a SSL-encrypted channel when they intended to use one.

**Comment by chefabien** [comment]
Hi Alan, I made the changes you requested. Hope it is what you expected.

**Comment by alanxz** [comment]
Squashed, and merged into the master branch. Thanks for the contribution.

**Pull request closed without merging**